src/EasyNetQ.HostedService/Internals/HandlerRegistrar.cs

```diff
@@ -25,7 +32,38 @@ public IHandlerRegistration Add<T>(IMessageHandler<T> handler)
                     await _incomingMessageInterceptor.InterceptMessage(message, messageReceivedInfo, cancellationToken);
                 }
 
-                return await handler(message, messageReceivedInfo, cancellationToken);
+                using (var activity = _activitySource.StartActivity(TraceActivityName.Consume, ActivityKind.Consumer))
```

Comment:
As we discussed we would need to create the activity as the child of the relevant sending span by reading the context propagation headers.

See https://github.com/jbogard/NServiceBus.Extensions.Diagnostics/blob/master/src/NServiceBus.Extensions.Diagnostics/IncomingPhysicalMessageDiagnostics.cs for a relevant implementation

Other links:
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/examples/MicroserviceExample/Utils/Messaging
https://www.w3.org/TR/trace-context/